pkg/proc/internal/ebpf/trace_probe/trace.bpf.c

```diff
@@ -0,0 +1,165 @@
+#include "trace.bpf.h"
+#include <string.h>
+
+#define STRING_KIND 24
+
+// parse_string_param will parse a string parameter. The parsed value of the string
+// will be put into param->deref_val. This function expects the string struct
+// which contains a pointer to the string and the length of the string to have
+// already been read from memory and passed in as param->val.
+__always_inline
+int parse_string_param(struct pt_regs *ctx, function_parameter_t *param) {
+    u64 str_len;
+    size_t str_addr;
+
+    memcpy(&str_addr, param->val, sizeof(str_addr));
+    memcpy(&str_len, param->val + sizeof(str_addr), sizeof(str_len));
+    param->daddr = str_addr;
+
+    if (str_addr != 0) {
+        if (str_len > 0x30) {
+            str_len = 0x30;
+        }
+        int ret = bpf_probe_read_user(&param->deref_val, str_len, (void *)(str_addr));
+        if (ret < 0) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+__always_inline
+int parse_param_stack(struct pt_regs *ctx, function_parameter_t *param) {
+    long ret;
+    size_t addr = ctx->sp + param->offset;
+    ret = bpf_probe_read_user(&param->val, param->size, (void *)(addr));
+    if (ret < 0) {
+        return 1;
+    }
+    return 0;
+}
+
+__always_inline
+void get_value_from_register(struct pt_regs *ctx, void *dest, int reg_num) {
+    switch (reg_num) {
```

Comment:
Are the other registers not there?

Reply:
They are, I was just being lazy and selfishly implemented only the ones I needed ;). Added the rest.